core/controlplane/config/templates/cloud-config-etcd

```diff
@@ -153,7 +153,7 @@ coreos:
         After=network.target
 
         [Service]
-        Type=oneshot
+        Restart=on-failure
```

Comment:
Good point! But just changing this to `Type=simple`(as you removed `Type=oneshot`) results in systemd to not wait until `etcdadm-reconfigure` finishes.

If you really wanted `etcadm-reconfigure` to be retried automatically, would you mind achieving it like `/opt/bin/retry 5 /opt/bin/etcdadm reconfigure` instead?

FYI, you can see similar implementations are done in `cloud-config-controller` for `install-kube-system` and `apply-kube-aws-plugins`.